app/controllers/forms_controller.rb

```diff
@@ -16,4 +17,24 @@ def update
       render "question_flow/#{step}"
     end
   end
+
+protected
+
+  def check_early_eligibility
+    if gross_income_details_valid?
+      session_data["api_result"] = CfeService.call(session_data, early_eligibility: true)
+      @early_eligibility_result = EarlyEligibility.new(session_data)
```

Comment:
This assignment won't do much, as it'll get wiped when we redirect the user to a new page. What you'll need is for the _next_ page to be rendered to know that it needs to show something about early eligibility. My thinking (based on the untested assumption that this early eligibility indicator will make more sense as a standalone screen than a widget above a form, as it contains such important information), was that you could add a new screen into the flow just like you'd add any other screen, called `post_gross_income_early_eligibility`. You'd add a new Steps::PostGrossIncomeEarlyEligibilitySection containing the logic about whether to skip that screen, and that logic would be "skip it if the early eligibility CFE result is 'eligible'".

The helpful thing about _that_ is that that new screen can then easily contain a form, like all the other screens. And that form can say "Would you like to skip to the end or not?", and we can store the user's answer in the session, just like for every other form, and then because that answer is stored in the session, we can use it to determine, in the other Steps::BlahSection classes, via Steps::Logic, whether to display or skip all the _subsequent_ screens in the flow. i.e. we can re-use our existing navigation system to handle the navigation options we give to the user after an early eligibility result.